### PR TITLE
CP VC changes to fpgadiag 1.4.0 branch

### PR DESCRIPTION
**Pull request by anandaravuri**
- mactest & fpgastats tools used in PAC card D5005.
- CP changes to support network card testing.

### CODE REVIEW

File: tools/extra/fpgadiag/common.py [hunk 1]
```diff
@@ -221,19 +261,87 @@ def get_port_list(self, argport, total):
...
-    parser.add_argument('--bus', '-B', type=hexint,
-                        help='Bus number of PCIe device')
-    parser.add_argument('--device', '-D', type=hexint,
-                        help='Device number of PCIe device')
-    parser.add_argument('--function', '-F', type=hexint,
-                        help='Function number of PCIe device')
+    parser.add_argument('--segment')
+    parser.add_argument('--bus')
+    parser.add_argument('--device')
+    parser.add_argument('--function')
```
Comment (jelonanX) [inline tools/extra/fpgadiag/common.py]: help messages might be useful here. Not sure why it was taken out

File: tools/extra/fpgadiag/fpgastats.py [hunk 2]
```diff
@@ -27,20 +27,27 @@
 
 from __future__ import print_function
 from common import exception_quit, FpgaFinder, COMMON
-from common import hexint
+from common import convert_argument_str2hex
```
Comment (jelonanX) [inline tools/extra/fpgadiag/fpgastats.py]: if it's from the same library, couldn't we import it with one line? (just a style thing)

File: tools/extra/fpgadiag/mactest.py [hunk 3]
```diff
@@ -30,20 +30,30 @@
 import argparse
 import binascii
 import sys
+import glob
 from common import FpgaFinder, exception_quit
-from common import COMMON, hexint
+from common import COMMON, convert_argument_str2hex
```
Comment (jelonanX) [inline tools/extra/fpgadiag/mactest.py]: if it's from the same library, couldn't we import it with one line? (just a style thing)

File: tools/extra/fpgadiag/common.py [hunk 4]
```diff
@@ -35,23 +36,53 @@
...
 import stat
 import struct
+import mmap
+
 
 PATTERN = (r'.*(?P<segment>\w{4}):(?P<bus>\w{2}):'
            r'(?P<dev>\w{2})\.(?P<func>\d).*')
 
-BDF_PATTERN = re.compile(PATTERN)
+BDF_PATTERN  = re.compile(PATTERN)
```
Comment (tswhison) [inline tools/extra/fpgadiag/common.py]: Remove extra space char before =

File: tools/extra/fpgadiag/common.py [hunk 5]
```diff
@@ -66,7 +97,7 @@ def __init__(self, segment, bus, dev, func):
 
     def read_bdf(self, path):
         symlink = os.readlink(path)
-        m = BDF_PATTERN.match(symlink)
+        m = BDF_PATTERN .match(symlink)
```
Comment (tswhison) [inline tools/extra/fpgadiag/common.py]: Remove space before .

File: tools/extra/fpgadiag/common.py (hunk 4 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/common.py]: fixed

File: tools/extra/fpgadiag/common.py (hunk 5 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/common.py]: fixed

File: tools/extra/fpgadiag/common.py (hunk 1 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/common.py]: fixed

File: tools/extra/fpgadiag/fpgastats.py (hunk 2 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/fpgastats.py]: fixed

File: tools/extra/fpgadiag/mactest.py (hunk 3 above)
Comment (anandaravuri) [reply tools/extra/fpgadiag/mactest.py]: fixed

**Comment by ibarraza** [comment]
It looks like the previous PR also added the fvlbypass option to fpgadiag, which is not included in this version.

File: tools/extra/fpgadiag/mactest.py [hunk 6]
```diff
@@ -31,11 +31,11 @@
 import binascii
 import sys
 import glob
-from common import FpgaFinder, exception_quit
-from common import COMMON, convert_argument_str2hex
+from common import FpgaFinder, exception_quit, COMMON, hexint
+
+SYSF_IF = '/sys/class/net'
+DIVIDE = '-' * 80
```
Comment (ibarraza) [inline tools/extra/fpgadiag/mactest.py]: Lines 85 and 130 also need to be updated.

Comment (anandaravuri) [reply tools/extra/fpgadiag/mactest.py]: Fixed

File: tools/extra/fpgadiag/mactest.py [hunk 7]
```diff
@@ -31,11 +31,11 @@
...
 
-sys_if = '/sys/class/net'
-divide = '-' * 80
 FVL_SIDE = 1
```
Comment (r-rojo) [inline tools/extra/fpgadiag/mactest.py]: Should we get rid of code names? Maybe this can be HOST_SIDE?

Comment (anandaravuri) [reply tools/extra/fpgadiag/mactest.py]: Fixed

**Pull request merged**